src/nvim/memory.c

```diff
@@ -512,7 +512,14 @@ void *xmemdup(const void *data, size_t len)
 bool strequal(const char *a, const char *b)
   FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
 {
-  return (a == NULL && b == NULL) || (a && b && strcmp(a, b) == 0);
+  return strcmp(a ? a : "", b ? b : "") == 0;
```

Comment:
The logic is not quite the same. Previously, `a==b` if _both_ are NULL, but _not_ if one is NULL and one is `""`.

Reply:
I am aware of this technical change, but functionally it still behaves the same everywhere it's called.
And imo, this behavior makes more sense if we don't want to discriminate between 0-length strings.
Can change this back if required, though.

Reply:
I wouldn't change the semantics. E.g. for environment variables, NULL means something different than empty string. `strequal` could no longer be used for that case.

If we want to change this we need to think about all call sites and should be done in a separate PR.

Reply:
That's valid. How do you think `strnequal` (not `strequal`) should behave in that case for `n == 0` if `a == NULL` and `b != NULL`?

Reply:
I changed `strequal` back to what it was before and made `strncmp` mimic its behavior as well. Lmk if you think it looks fine.